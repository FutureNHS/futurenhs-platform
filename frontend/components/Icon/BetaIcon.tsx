import React from "react";

interface Props {
  className?: string;
}

const BetaIcon = ({ className }: Props) => (
  <div className={`icon-wrapper ${className || ""}`}>
    <svg
      width="50"
      height="24"
      viewBox="0 0 50 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M2 0C0.895431 0 0 0.89543 0 2V22C0 23.1046 0.895432 24 2 24H48C49.1046 24 50 23.1046 50 22V2C50 0.895431 49.1046 0 48 0H2ZM9.13477 6.66211V16H11.9648C12.5573 16 13.0426 15.9749 13.4209 15.9248C13.8037 15.8747 14.1706 15.7448 14.5215 15.5352C14.7767 15.3848 14.9954 15.2025 15.1777 14.9883C15.36 14.7695 15.4967 14.5257 15.5879 14.2568C15.679 13.988 15.7246 13.71 15.7246 13.4229C15.7246 13.0856 15.6722 12.7871 15.5674 12.5273C15.4626 12.263 15.3122 12.0329 15.1162 11.8369C14.9202 11.6364 14.6855 11.4769 14.4121 11.3584C14.1432 11.2354 13.8584 11.1579 13.5576 11.126C14.1637 10.9984 14.6195 10.766 14.9248 10.4287C15.2347 10.0869 15.3896 9.61523 15.3896 9.01367C15.3896 8.13411 15.0752 7.52344 14.4463 7.18164C13.8174 6.83529 13.0404 6.66211 12.1152 6.66211H9.13477ZM11.9307 14.5713H11.0215V11.8438H12.0605C12.571 11.8438 12.9971 11.9531 13.3389 12.1719C13.6807 12.3861 13.8516 12.7324 13.8516 13.2109C13.8516 13.7031 13.6647 14.054 13.291 14.2637C12.9219 14.4688 12.4684 14.5713 11.9307 14.5713ZM11.917 10.4424H11.0215V8.10449H11.9785C12.4388 8.10449 12.8102 8.19564 13.0928 8.37793C13.3753 8.55566 13.5166 8.861 13.5166 9.29395C13.5166 9.71322 13.3639 10.0094 13.0586 10.1826C12.7578 10.3558 12.3773 10.4424 11.917 10.4424ZM23.1416 16V14.4756H19.4434V11.9805H22.7793V10.4561H19.4434V8.2002H23.0869V6.66211H17.5566V16H23.1416ZM28.6309 16H26.7441V8.2002H24.0781V6.66211H31.3174V8.2002H28.6309V16ZM39.0215 16H41.0791L37.3398 6.66211H35.2412L31.5977 16H33.5459L34.3457 13.874H38.208L39.0215 16ZM36.4307 9.04102L37.6338 12.3428H34.9199L36.041 9.19824C36.0684 9.11165 36.1003 8.99772 36.1367 8.85645C36.1732 8.71061 36.2096 8.56706 36.2461 8.42578C36.3418 8.76302 36.4033 8.9681 36.4307 9.04102Z"
        fill="#4C6272"
      />
    </svg>
  </div>
);

export default BetaIcon;
