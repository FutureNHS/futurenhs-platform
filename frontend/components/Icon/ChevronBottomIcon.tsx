import React from "react";

interface Props {
  className?: string;
  title?: string;
  description?: string;
}

const ChevronBottomIcon = ({ className, title, description }: Props) => (
  <div className={`icon-wrapper ${className || ""}`}>
    <svg
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      aria-labelledby="chevronBottomIconId chevronBottomIconDesc"
    >
      {title ? <title id="chevronBottomIconId">{title}</title> : ""}
      {description ? <desc id="chevronBottomIconDesc">{description}</desc> : ""}
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M11.5 16.5C11.1 16.5 10.8333 16.3714 10.5667 16.1143L3.9 9.68571C3.36667 9.17143 3.36667 8.4 3.9 7.88571C4.43333 7.37143 5.23333 7.37143 5.76667 7.88571L11.5 13.4143L17.2333 7.88571C17.7667 7.37143 18.5667 7.37143 19.1 7.88571C19.6333 8.4 19.6333 9.17143 19.1 9.68571L12.4333 16.1143C12.1667 16.3714 11.9 16.5 11.5 16.5V16.5Z"
        fill="#425462"
      />
    </svg>
  </div>
);

export default ChevronBottomIcon;
