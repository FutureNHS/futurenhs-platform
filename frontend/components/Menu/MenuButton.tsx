import React, { FC, ComponentPropsWithoutRef } from "react";

import classNames from "classnames";
import styled from "styled-components";

import { Tooltip } from "../Tooltip";

type Background = "light" | "dark";
interface MenuProps extends ComponentPropsWithoutRef<"button"> {
  hiddenUntilHover: boolean;
  menuOpen: boolean;
  setMenuOpen: (open: boolean) => void;
  dataCy?: string;
  background: Background;
}

const MenuButton: FC<MenuProps> = ({
  className,
  hiddenUntilHover,
  menuOpen,
  setMenuOpen,
  dataCy,
  background,
  ...rest
}) => {
  return (
    <Tooltip tooltip="Options">
      <button
        className={classNames(
          { open: menuOpen },
          { hidden: hiddenUntilHover },
          { light: background === "light" },
          className
        )}
        onClick={() => setMenuOpen(!menuOpen)}
        aria-label="Options"
        data-cy={dataCy}
        {...rest}
      />
    </Tooltip>
  );
};

const StyledMenuButton = styled(MenuButton)`
  opacity: 1;
  color: ${({ theme }) => theme.colorNhsukGrey1};
  background-color: ${({ theme }) => theme.colorNhsukGrey4};
  &.light {
    background-color: ${({ theme }) => theme.colorNhsukGrey5};
  }
  &.hidden {
    opacity: 0;
  }
  :hover {
    opacity: 1;
    color: ${({ theme }) => theme.colorNhsukWhite};
    &.hidden {
      color: ${({ theme }) => theme.colorNhsukWhite};
    }
    background-color: ${({ theme }) => theme.colorShadeNhsukBlue35};
  }
  &.open,
  :active,
  :focus {
    opacity: 1;
    color: ${({ theme }) => theme.colorNhsukBlack};
    &.hidden {
      color: ${({ theme }) => theme.colorNhsukBlack};
    }
    background-color: ${({ theme }) => theme.colorNhsukYellow};
    outline: none;
    border-bottom: 2px solid black;
  }

  border-radius: 4px;
  border: none;
  cursor: pointer;
  display: inline-block;
  font-size: 16px;
  height: 100%;
  line-height: 24px;
  padding: 0;
  position: relative;

  .icon-wrapper {
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;

export default StyledMenuButton;
