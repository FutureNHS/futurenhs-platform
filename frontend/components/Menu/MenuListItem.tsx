import React, { ComponentPropsWithoutRef, FC, ReactNode } from "react";

import Link from "next/link";
import styled from "styled-components";

type Href = string;
type OnClickFn = () => void;

interface ListItemProps {
  className?: string;
  children: ReactNode;
  handler: Href | OnClickFn;
}

const ListItem = ({ className, children, handler }: ListItemProps) => {
  return (
    <li className={className}>
      {typeof handler === "function" ? (
        <button onClick={handler}>{children}</button>
      ) : (
        <Link href={handler} passHref>
          <a>{children}</a>
        </Link>
      )}
    </li>
  );
};

const StyledListItem = styled(ListItem)`
  box-shadow: inset 0px -1px 0px #e8edee;
  list-style: none;
  margin: 0;

  a,
  button {
    align-items: center;
    border: none;
    display: flex;
    justify-content: space-between;
    padding: 12px 16px;
    width: 100%;
  }

  button {
    background: ${({ theme }) => theme.colorNhsukWhite};
    font-size: 19px;
    line-height: 28px;
    text-align: left;
    border-radius: 4px;
    border: none;
    outline: 4px solid transparent;
  }

  .icon-wrapper {
    align-items: center;
    display: flex;
    justify-content: center;
  }

  ${({ theme }) => `
   button {
     &:hover {
        color: ${theme.colorNhsukWhite};
        &.hidden {
          color: ${theme.colorNhsukWhite};
        }
        background-color: ${theme.colorShadeNhsukBlue35};
      }
      &:active, :focus {
        color: ${theme.colorNhsukBlack};
        &.hidden {
          color: ${theme.colorNhsukBlack};
        }
        border:none;
        background-color: ${theme.colorNhsukYellow};
        box-shadow: 0 -2px transparent, 0 4px #212b32;
      }
   }

    a, button {
      color: ${theme.colorNhsukBlue};
      text-decoration: none;

      &:hover:not(:active) {
        color: ${theme.colorNhsukWhite};
        background-color: ${theme.colorShadeNhsukBlue35};
      }
    }
  `}
`;

const StyledTitle = styled.div`
  padding-left: 20px;
  flex-grow: 1;
`;

export type MenuItem = {
  title: string;
  icon: ReactNode;
  handler: Href | OnClickFn;
  dataCy?: string;
};

interface MenuListItemProps extends ComponentPropsWithoutRef<"li"> {
  title: string;
  handler: Href | OnClickFn;
  dataCy?: string;
}

const MenuListItem: FC<MenuListItemProps> = ({
  title,
  children,
  dataCy,
  ...props
}) => {
  return (
    <StyledListItem {...props}>
      {children}
      {title && <StyledTitle data-cy={dataCy}>{title}</StyledTitle>}
    </StyledListItem>
  );
};

export default MenuListItem;
