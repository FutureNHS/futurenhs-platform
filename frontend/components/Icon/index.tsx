import BetaIcon from "./BetaIcon";
import CrossIcon from "./CrossIcon";
import DashboardIcon from "./DashboardIcon";
import DeleteIcon from "./DeleteIcon";
import EditIcon from "./EditIcon";
import FileIcon from "./FileIcon";
import FnhsLogoIcon from "./FnhsLogoIcon";
import HelpIcon from "./HelpIcon";
import LockIcon from "./LockIcon";
import LogOutIcon from "./LogOutIcon";
import MeatballIcon from "./MeatballIcon";
import MinusIcon from "./MinusIcon";
import MoveIcon from "./MoveIcon";
import NhsukLogoIcon from "./NhsukLogoIcon";
import NotificationsIcon from "./NotificationsIcon";
import OGLIcon from "./OGLIcon";
import PlusIcon from "./PlusIcon";
import TickIcon from "./TickIcon";
import UploadIcon from "./UploadIcon";
import UserIcon from "./UserIcon";
import WorkspacesIcon from "./WorkspacesIcon";

export {
  BetaIcon,
  CrossIcon,
  DashboardIcon,
  DeleteIcon,
  EditIcon,
  FileIcon,
  FnhsLogoIcon,
  HelpIcon,
  LockIcon,
  LogOutIcon,
  MeatballIcon,
  MinusIcon,
  MoveIcon,
  NhsukLogoIcon,
  NotificationsIcon,
  OGLIcon,
  PlusIcon,
  TickIcon,
  UploadIcon,
  UserIcon,
  WorkspacesIcon,
};
