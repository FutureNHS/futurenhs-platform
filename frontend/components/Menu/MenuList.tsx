import React, { FC, ComponentPropsWithoutRef } from "react";

import classNames from "classnames";
import styled from "styled-components";

import { MenuListItem } from ".";
import { MenuItem } from "./MenuListItem";

const Container = styled.div`
  z-index: 1000;
  width: 320px;
  border-radius: 4px;

  top: 32px;
  right: 0px;

  &.hidden {
    top: 0px;
    left: 32px;
  }

  ${({ theme }) => `
    position: absolute;
    background-color: ${theme.colorNhsukWhite};
    filter: drop-shadow(0px 0px 10px ${theme.nhsukBoxShadow});
  `}

  span {
    font-weight: 700;
    align-self: center;
  }

  ul {
    padding: 0;
    margin: 0;
    width: 320px;

    li:first-child a,
    li:first-child button {
      border-radius: 4px 4px 0 0;
    }

    li:last-child a,
    li:last-child button {
      border-radius: 0 0 4px 4px;
    }
  }
`;

interface Props extends ComponentPropsWithoutRef<"div"> {
  children?: MenuItem[];
  hiddenUntilHover: boolean;
}

const MenuList: FC<Props> = ({ children, hiddenUntilHover, className }) => {
  return (
    <Container className={classNames({ hidden: hiddenUntilHover }, className)}>
      <ul>
        {children?.map((item) => {
          const { icon, ...props } = item;
          return (
            <MenuListItem key={props.title} {...props} dataCy={props.dataCy}>
              {icon}
            </MenuListItem>
          );
        })}
      </ul>
    </Container>
  );
};

export default MenuList;
