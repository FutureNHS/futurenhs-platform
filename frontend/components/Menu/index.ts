import Menu from "./Menu";
import MenuButton from "./MenuButton";
import MenuListItem, { MenuItem } from "./MenuListItem";

export { Menu, MenuButton, MenuListItem };
export type { MenuItem };
