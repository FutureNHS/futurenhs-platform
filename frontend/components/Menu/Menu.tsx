import React, { FC, useEffect, useState } from "react";

import styled from "styled-components";

import { MenuButton } from ".";
import { MeatballIcon } from "../Icon";
import MenuList from "./MenuList";
import { MenuItem } from "./MenuListItem";

interface Props {
  hiddenUntilHover?: boolean;
  items: MenuItem[];
  folderId: string;
}

const Container = styled.div`
  align-items: center;
  position: relative;
  height: 100%;
  border-radius: 4px;
  margin-left: 7px;

  .tooltip {
    left: 36px;
    top: -7px;
  }

  ${({ theme }) => `
    .folder-menu-item {
      display: none;

      a {
        color: ${theme.colorNhsukWhite};
        &:focus {
          color: ${theme.colorNhsukBlack};
        }
      }
    }

    @media (min-width: ${theme.mqBreakpoints.largeDesktop}) {
      justify-content: space-between;
      display: flex;

      .folder-menu-item {
        display: flex;
      }
    }
  `}
`;

const Menu: FC<Props> = ({ hiddenUntilHover, items }) => {
  const [menuOpen, setMenuOpen] = useState(false);
  const container = React.useRef<HTMLDivElement>(null);

  useEffect(() => {
    const pageClickEvent = ({ target }: MouseEvent) => {
      if (!container.current?.contains(target as Node)) {
        setMenuOpen(false);
      }
    };

    if (menuOpen) {
      window.addEventListener("click", pageClickEvent);
    }

    return () => window.removeEventListener("click", pageClickEvent);
  }, [menuOpen]);

  return (
    <>
      <Container ref={container}>
        <MenuButton
          menuOpen={menuOpen}
          setMenuOpen={setMenuOpen}
          hiddenUntilHover={hiddenUntilHover || false}
        >
          <MeatballIcon />
        </MenuButton>
        {menuOpen && (
          <MenuList hiddenUntilHover={hiddenUntilHover || false}>
            {items}
          </MenuList>
        )}
      </Container>
    </>
  );
};

export default Menu;
