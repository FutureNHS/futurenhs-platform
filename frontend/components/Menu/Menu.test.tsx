import React from "react";

import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import theme from "../../lib/fixtures/theme.json";
import { UploadIcon } from "../Icon";
import Menu from "./Menu";

describe("<FolderMenu/>", () => {
  const workspaceId = "workspace-id";
  const folderId = "folder-id";

  test("takes a snapshot, as used in the nav", () => {
    const workspaceId = "workspace-id";
    const folderId = "folder-id";
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Menu
          hiddenUntilHover={true}
          folderId="f1"
          items={[
            {
              title: "Upload file to this folder",
              icon: <UploadIcon />,
              href: `/workspaces/${workspaceId}/folders/${folderId}/upload-file`,
            },
          ]}
        />
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("takes a snapshot, as used in a page", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Menu
          folderId="f1"
          items={[
            {
              title: "Upload file to this folder",
              icon: <UploadIcon />,
              href: `/workspaces/${workspaceId}/folders/${folderId}/upload-file`,
            },
          ]}
        />
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
