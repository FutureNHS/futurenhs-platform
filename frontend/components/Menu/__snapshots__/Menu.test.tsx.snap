// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu takes a snapshot, as used in a page 1`] = `
<DocumentFragment>
  .c2 {
  display: inline-block;
  position: relative;
}

.c5 {
  opacity: 0;
  position: absolute;
  z-index: 5;
  border-radius: 4px;
  background-color: rgb(76,98,114);
  padding: 5px;
  font-weight: 400;
  font-size: 18px;
  color: rgb(255,255,255);
}

.c1:hover .c4 {
  opacity: 1;
  -webkit-transition: opacity 0s linear 0.5s;
  transition: opacity 0s linear 0.5s;
}

.c5::after {
  position: absolute;
  bottom: 33%;
  left: -10%;
  margin-left: -2px;
  border-width: 5px;
  border-style: solid;
  border-color: transparent rgb(76,98,114) transparent transparent;
  content: " ";
}

.c3 {
  opacity: 1;
  color: rgb(76,98,114);
  background-color: rgb(216,221,224);
  border-radius: 4px;
  border: none;
  cursor: pointer;
  display: inline-block;
  font-size: 16px;
  height: 100%;
  line-height: 24px;
  padding: 0;
  position: relative;
}

.c3.light {
  background-color: rgb(240,244,245);
}

.c3.hidden {
  opacity: 0;
}

.c3:hover {
  opacity: 1;
  color: rgb(255,255,255);
  background-color: rgb(0,61,120);
}

.c3:hover.hidden {
  color: rgb(255,255,255);
}

.c3.open,
.c3:active,
.c3:focus {
  opacity: 1;
  color: rgb(33,43,50);
  background-color: rgb(255,235,59);
  outline: none;
  border-bottom: 2px solid black;
}

.c3.open.hidden,
.c3:active.hidden,
.c3:focus.hidden {
  color: rgb(33,43,50);
}

.c3 .icon-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  height: 100%;
  border-radius: 4px;
  margin-left: 7px;
}

.c0 .tooltip {
  left: 36px;
  top: -7px;
}

@media (min-width:990px) {
  .c0 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <button
        aria-label="Options"
        class="c3"
      >
        <div
          class="icon-wrapper "
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M17 12a2 2 0 104 0 2 2 0 00-4 0zm-5 2a2 2 0 110-4 2 2 0 010 4zm-7 0a2 2 0 110-4 2 2 0 010 4z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
      <span
        class="c4 c5 tooltip"
      >
        Options
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Menu takes a snapshot, as used in the nav 1`] = `
<DocumentFragment>
  .c2 {
  display: inline-block;
  position: relative;
}

.c5 {
  opacity: 0;
  position: absolute;
  z-index: 5;
  border-radius: 4px;
  background-color: rgb(76,98,114);
  padding: 5px;
  font-weight: 400;
  font-size: 18px;
  color: rgb(255,255,255);
}

.c1:hover .c4 {
  opacity: 1;
  -webkit-transition: opacity 0s linear 0.5s;
  transition: opacity 0s linear 0.5s;
}

.c5::after {
  position: absolute;
  bottom: 33%;
  left: -10%;
  margin-left: -2px;
  border-width: 5px;
  border-style: solid;
  border-color: transparent rgb(76,98,114) transparent transparent;
  content: " ";
}

.c3 {
  opacity: 1;
  color: rgb(76,98,114);
  background-color: rgb(216,221,224);
  border-radius: 4px;
  border: none;
  cursor: pointer;
  display: inline-block;
  font-size: 16px;
  height: 100%;
  line-height: 24px;
  padding: 0;
  position: relative;
}

.c3.light {
  background-color: rgb(240,244,245);
}

.c3.hidden {
  opacity: 0;
}

.c3:hover {
  opacity: 1;
  color: rgb(255,255,255);
  background-color: rgb(0,61,120);
}

.c3:hover.hidden {
  color: rgb(255,255,255);
}

.c3.open,
.c3:active,
.c3:focus {
  opacity: 1;
  color: rgb(33,43,50);
  background-color: rgb(255,235,59);
  outline: none;
  border-bottom: 2px solid black;
}

.c3.open.hidden,
.c3:active.hidden,
.c3:focus.hidden {
  color: rgb(33,43,50);
}

.c3 .icon-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  height: 100%;
  border-radius: 4px;
  margin-left: 7px;
}

.c0 .tooltip {
  left: 36px;
  top: -7px;
}

@media (min-width:990px) {
  .c0 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <button
        aria-label="Options"
        class="hidden c3"
      >
        <div
          class="icon-wrapper "
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M17 12a2 2 0 104 0 2 2 0 00-4 0zm-5 2a2 2 0 110-4 2 2 0 010 4zm-7 0a2 2 0 110-4 2 2 0 010 4z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
      <span
        class="c4 c5 tooltip"
      >
        Options
      </span>
    </div>
  </div>
</DocumentFragment>
`;
